Permission denied with Deno v0.36.0

Thank you for this action! 

As the title says, my CI pipeline errored on `deno test` with the following, using 3.6:

`/home/runner/work/_temp/476f2026-a03c-44ac-be96-866004849968.sh: line 1: /opt/hostedtoolcache/deno/0.36.0/x64/deno: Permission denied
##[error]Process completed with exit code 126.`

See [this run](https://github.com/olaven/serialize-xml/runs/558064208) for an example. 

__note__: updating to 3.8 fixed the issue. 

Have a nice day! 😄

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -2,7 +2,7 @@ name: ci
 on: [push, pull_request]
 jobs:
   run:
-    name: Run on ${{ matrix.os }}
+    name: Deno v${{ matrix.deno }} in ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -7,6 +7,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macOS-latest]
+        deno: ["0.20.0", "0.38.0", "0.x"]
     steps:
       - uses: actions/checkout@master
 
@@ -24,7 +25,7 @@ jobs:
       - name: Test
         run: npm test
 
-      - name: Set up Deno
+      - name: Set up Deno v${{ matrix.deno }}
         uses: ./
 
       - name: Run deno

--- a/__tests__/installer.test.ts
+++ b/__tests__/installer.test.ts
@@ -63,9 +63,9 @@ describe("installer tests", () => {
     expect(fs.existsSync(path.join(denoDir, `deno${EXTENSION}`))).toBe(true);
   }, 100000);
 
-  it("Acquires version of deno which less then v0.35.0", async () => {
-    await installer.getDeno("v0.35.0");
-    const denoDir = path.join(toolDir, "deno", "0.35.0", os.arch());
+  it("Acquires version of deno which less then v0.38.0", async () => {
+    await installer.getDeno("v0.20.0");
+    const denoDir = path.join(toolDir, "deno", "0.20.0", os.arch());
 
     expect(fs.existsSync(`${denoDir}.complete`)).toBe(true);
     expect(fs.existsSync(path.join(denoDir, `deno${EXTENSION}`))).toBe(true);

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -47,11 +47,11 @@ import * as uuidV4 from "uuid";
 import { HttpClient } from "typed-rest-client/HttpClient";
 
 function getDenoArch(version: string): Arch {
-  return semver.lte(version, "0.35.0") ? "x64" : "x86_64";
+  return semver.lte(version, "0.38.0") ? "x64" : "x86_64";
 }
 function getDenoPlatform(version: string): Platform {
   const platform = os.platform();
-  const isLessThenV35 = semver.lte(version, "0.35.0");
+  const isLessThenV35 = semver.lte(version, "0.38.0");
 
   let rtv: Platform | null = null;
   if (platform === "darwin") rtv = isLessThenV35 ? "osx" : "apple-darwin";
@@ -135,11 +135,11 @@ export async function getAvailableVersions() {
 }
 
 export function getDownloadUrl(version: string): string {
-  // The old release file only keep to Deno v0.35.0
+  // The old release file only keep to Deno v0.38.0
   const platform = getDenoPlatform(version);
   const arch = getDenoArch(version);
   let filename: string;
-  if (semver.lte(version, "0.35.0")) {
+  if (semver.lte(version, "0.38.0")) {
     const extName = process.platform === "win32" ? "zip" : "gz";
     filename = `deno_${platform}_${arch}.${extName}`;
   } else {
@@ -154,7 +154,7 @@ export async function extractDenoArchive(
   archiveFilepath: string
 ): Promise<string> {
   let extPath = "";
-  if (semver.lte(version, "0.35.0")) {
+  if (semver.lte(version, "0.38.0")) {
     if (process.platform === "win32") {
       extPath = await tc.extractZip(archiveFilepath);
     } else {

